File: Taller 3/Python/control.py

class Control:

    def __init__(self):
      self._tv = None


    #------------------------------
    def turnOff(self):
        self._tv.turnOff()   
    
    def turnOn(self):
        self._tv.turnOn()
    
    #......
    def canalUp(self):
        self._tv.canalUp()
    
    def canalDown(self):
        self._tv.canalDown()
    
    #......
    def volumenUp(self):
        self._tv.volumenUp()
    
    def volumenDown(self):
        self._tv.volumenDown()
    
    #......
    def setCanal(self, canal):
        self._tv.setCanal(canal)
    
    def setVolumen(self, volumen):
        self._tv.setVolumen(volumen)
    
    #------------------------------
    

    #------------------------------
    def enlazar(self, tv):
        self._tv = tv
        tv.setControl(self)
    
    #------------------------------

    def setTv(self, tv):
        self._tv = tv
    

    def getTv(self):
        return self._tv
    

File: Taller 3/Java/Control.java
package taller3.televisores;

public class Control {
    private TV tv;

    // ------------------------------
    public void turnOff() {
        this.tv.turnOff();
    }

    public void turnOn() {
        this.tv.turnOn();
    }

    // ......
    public void canalUp() {
        this.tv.canalUp();
    }

    public void canalDown() {
        this.tv.canalDown();
    }

    // ......
    public void volumenUp() {
        this.tv.volumenUp();
    }

    public void volumenDown() {
        this.tv.volumenDown();
    }

    // ......
    public void setCanal(int canal) {
        this.tv.setCanal(canal);
    }

    public void setVolumen(int volumen) {
        this.tv.setVolumen(volumen);
    }
    // ------------------------------

    // ------------------------------
    public void enlazar(TV tv) {
        this.tv = tv;
        tv.setControl(this);
    }
    // ------------------------------

    public void setTv(TV tv) {
        this.tv = tv;
    }

    public TV getTv() {
        return this.tv;
    }
}